packages/grid/x-grid/src/tests/filtering.XGrid.test.tsx

```diff
@@ -224,4 +226,85 @@ describe('<XGrid /> - Filter', () => {
     setProps({ filterModel: { items: [] } });
     expect(getColumnValues()).to.deep.equal(['Nike', 'Adidas', 'Puma']);
   });
+
+  describe('Server', () => {
+    it('should refresh the filter panel when adding filters', () => {
+      function loadServerRows(commodityFilterValue) {
+        const serverRows = [
+          { id: '1', commodity: 'rice' },
+          { id: '2', commodity: 'soybeans' },
+          { id: '3', commodity: 'milk' },
+          { id: '4', commodity: 'wheat' },
+          { id: '5', commodity: 'oats' },
+        ];
+
+        return new Promise<RowModel[]>((resolve) => {
+          setTimeout(() => {
+            if (!commodityFilterValue) {
+              resolve(serverRows);
+              return;
+            }
+            resolve(
+              serverRows.filter(
+                (row) => row.commodity.toLowerCase().indexOf(commodityFilterValue) > -1,
+              ),
+            );
+          }, Math.random() * 500 + 100); // simulate network latency
+        });
+      }
+
+      function AddServerFilterGrid() {
+        const [columns] = React.useState([{ field: 'commodity', width: 150 }]);
+        const [rows, setRows] = React.useState<RowModel[]>([]);
+        const [filterValue, setFilterValue] = React.useState();
+        const [loading, setLoading] = React.useState(false);
+
+        const onFilterChange = React.useCallback((params) => {
+          setFilterValue(params.filterModel.items[0].value);
+        }, []);
+
+        React.useEffect(() => {
+          let active = true;
+
+          (async () => {
+            setLoading(true);
+            const newRows = await loadServerRows(filterValue);
+
+            if (!active) {
+              return;
+            }
+
+            setRows(newRows);
+            setLoading(false);
+          })();
+
+          return () => {
+            active = false;
+          };
+        }, [filterValue]);
+
+        return (
+          <div style={{ height: 400, width: 400 }}>
+            <XGrid
+              rows={rows}
+              columns={columns}
+              filterMode="server"
+              onFilterModelChange={onFilterChange}
+              loading={loading}
+              state={{
+                preferencePanel: { open: true, openedPanelValue: PreferencePanelsValue.filters },
+              }}
+            />
+          </div>
+        );
+      }
+
+      render(<AddServerFilterGrid />);
+      const addButton = document.querySelector(`.MuiAddFilterButton`);
+      clock.tick(100);
+      fireEvent.click(addButton);
+      const filterForms = document.querySelectorAll(`.MuiDataGridFilterForm-root`);
+      expect(filterForms.length).to.eq(2);
```

Comment:
We should likely make this alias throw in the setup

```suggestion
      expect(filterForms.length).to.equal(2);
```

or maybe better

```suggestion
      expect(filterForms).to.have.length(2);
```